src/Chainweb/Pact/PactService.hs

```diff
@@ -1564,11 +1606,21 @@ execLookupPactTxs restorePoint txs
     go = getCheckpointer >>= \(!cp) -> case restorePoint of
       NoRewind _ ->
         liftIO $! V.mapM (_cpLookupProcessedTx cp) txs
-      DoRewind bh -> withDiscardedBatch $ do
-        let !t = Just (_blockHeight bh + 1, _blockHash bh)
-        withCheckpointerRewind Nothing t "lookupPactTxs" $ \_ ->
+      DoRewind parent -> withDiscardedBatch $ do
+        withCheckpointerRewind Nothing (Just $ ParentHeader parent) "lookupPactTxs" $ \_ ->
           liftIO $ Discard <$> V.mapM (_cpLookupProcessedTx cp) txs
 
+-- | Find the latest block stored in the checkpointer for which the respective
+-- block header is available in the block header database.
+--
+-- NOTE: the function doesn't take into consideration the block header stored in
+-- '_psParentHeader' in 'PactServiceM'. It is meant to be used during re-orgs
```

Comment:
> It is meant to be used during re-orgs

Is this true? It is used in initialization but also in `rewindTo`, so that doesn't sound like re-org only.

Reply:
yes, it is true -- but maybe the comment isn't clear. In both places it is called to resolve possible forks. It should only be called when it is acceptable (as during initialization) or desirable (during rewindTo) to discard the latest validated block. Calling this function when '_cpLatestBlock' is not a fork/orphan is inefficient. In particular when there are many local calls, calling this function would result in many redundant re-evaluations.